src/euphorie/client/browser/templates/locking_badge.pt

```diff
@@ -4,23 +4,45 @@
                  session here/session;
                "
                condition="webhelpers/use_locking_feature"
+               i18n:domain="euphorie"
 >
-  <a class="icon icon-lock${python:'-open' if not session.is_locked else None} pat-tooltip"
-     href="${here/absolute_url}/@@locking_menu#locking-menu"
-     data-pat-tooltip="source: ajax; position-list: tr"
-     i18n:domain="euphorie"
-  >
-    <tal:switch switch="locking_view/state">
-      <tal:case case="string:locked"
-                i18n:translate="label_locked"
-      >
-        Locked
-      </tal:case>
-      <tal:case case="default"
-                i18n:translate="label_unlocked"
-      >
-        Unlocked
-      </tal:case>
-    </tal:switch>
-  </a>
+  <tal:switch switch="locking_view/state">
+    <tal:case case="string:locked">
+      <a class="icon-lock text-button pat-tooltip"
+         href="${here/absolute_url}/@@locking_menu#locking-menu"
+         title="Unlock this assessment"
+         data-pat-tooltip="source: ajax; position-list: tr; trigger: click"
+         tal:define="
+           time session/last_locking_event/time;
+           time_iso python:webhelpers.timezoned_date(time);
+           language here/plone_portal_state/language;
+         "
+         i18n:translate="label_locked_time"
+      >Locked
+        <time class="pat-display-time"
+              datetime="${time_iso}"
+              data-pat-display-time="from-now: true; locale: ${language}"
+              i18n:name="locking_time"
+        >${time_iso}</time>
+      </a>
+    </tal:case>
+    <tal:case case="string:validated">
+      <a class="pat-select success locked text-button pat-tooltip"
+         href="${here/absolute_url}/@@locking_menu#locking-menu"
+         title="Unlock this assessment"
+         data-pat-tooltip="source: ajax; position-list: tr; trigger: click; source: ajax"
+         i18n:translate="label_validated"
+      >Validated
+      </a>
+    </tal:case>
+    <tal:case case="default">
+      <a class="icon-lock-open icon pat-tooltip"
+         href="${here/absolute_url}/@@locking_menu#locking-menu"
+         title="Lock this assessment"
+         data-pat-tooltip="source: ajax; position-list: tr; trigger: click"
+         i18n:translate="label_unlocked"
+      >Unlocked
+      </a>
+    </tal:case>
+  </tal:switch>
```

Comment:
Am I wrong, or the session can be both locked and validated?
I would keep this possibility at least.

Reply:
If it is validated, then it is always also locked. You cannot unlock it without invalidating it, and when you invalidate it, it is also unlocked. So the three possible states are unlocked, locked, and validated. From the validated state you can only go to the unlocked state. From the other states you can go anywhere.

Reply:
> If it is validated, then it is always also locked

I tend to disagree here... 
I also discussed that with @pilz and IIRC we agreed the two things are orthogonal.

When it is validated, it is not locked. It is just not editable.

Should we talk about that again?

Reply:
Also remember that we already have customers that use the locking, and for them nothing should change.

Reply:
unlocked -> locked -> validated (also not editable)
similar to private -> submitted -> published. 
There is no need to go from validated to unlocked. I find it more suitable to go from validated to locked because the whole path is an edge case and if it happens, I would like it to be as much controlled as possible. And therefore as explicit as possible. Each step back should be a conscious step. The owner of the organisation should invalidate (Use case that just someone else should validate again). And if the owner decides that he want to also change something, he should explicitly unlock (use case that a change is necessary).

Let me know if that helps.

Reply:
@pilz, will you take this to Cornelis to update the design?

Reply:
I've updated this PR so that after invalidating an assessment it is still locked, as discussed this morning.